kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/jdbc/JDBCStateStore.scala

```diff
@@ -0,0 +1,446 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.server.statestore.jdbc
+
+import java.io.{BufferedReader, InputStream, InputStreamReader}
+import java.sql.{Connection, PreparedStatement, ResultSet, SQLException}
+import java.util.{Locale, Properties}
+import java.util.stream.Collectors
+
+import scala.collection.mutable.ListBuffer
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.google.common.annotations.VisibleForTesting
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+
+import org.apache.kyuubi.{KyuubiException, Logging, Utils}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf._
+import org.apache.kyuubi.server.statestore.StateStore
+import org.apache.kyuubi.server.statestore.api.BatchMetadata
+import org.apache.kyuubi.server.statestore.jdbc.DatabaseType._
+
+class JDBCStateStore(conf: KyuubiConf) extends StateStore with Logging {
+  import JDBCStateStore._
+
+  private val dbType = DatabaseType.withName(conf.get(SERVER_STATE_STORE_JDBC_DATABASE_TYPE))
+  private val driverClassOpt = conf.get(SERVER_STATE_STORE_JDBC_DRIVER)
+  private val driverClass = dbType match {
+    case DERBY => driverClassOpt.getOrElse("org.apache.derby.jdbc.AutoloadedDriver")
+    case MYSQL => driverClassOpt.getOrElse("com.mysql.jdbc.Driver")
+    case CUSTOM => driverClassOpt.getOrElse(
```

Comment:
it seems `driverClassOpt ` is enough , the `dbType` is redundant ?

Reply:
dbType is also used for database adaptor